instrumentation-annotations-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/async/AsyncEndStrategy.java

```diff
@@ -11,9 +11,10 @@
 /**
  * Implementations of this interface describe how to compose over {@linkplain #supports(Class)
  * supported} asynchronous computation types and delay marking the operation as ended by calling
- * {@link Instrumenter#end(Context, Object, Object, Throwable)}.
+ * {@link Instrumenter#end(Context, Object, Object, Throwable)} or {@link
+ * AsyncEndHandler#handle(Context, Object, Object, Throwable)}.
  */
-public interface AsyncOperationEndStrategy {
+public interface AsyncEndStrategy {
```

Comment:
I don't think we should change these names, at least not in this PR, because

- it makes the PR larger and harder to review
- it is a public API and so we would need to consider / weigh / document the breaking change

Reply:
I have checked the `opentelemetry-java` and `opentelemetry-java-contrib`. There are no dependency on the AsyncOperationXXX classes. But if it's the public api, I would prefer to keep the original declaration too. I think we can do when upgrading the major version.

Reply:
we don't need to wait for a major version, but I'd prefer to keep the number of changes in this PR down to make it easier to review since it's already been a long time in the waiting

Reply:
ok

Reply:
revert the rename of AsyncOperationEndXXX